### ISSUE TITLE
NixOS fails to build: iproute2-6.5.0/etc': No such file or directory

### ISSUE THREAD

**Issue by nadir-yttrium** [NixOS/nixpkgs#263445]
### Describe the bug
I was just trying to upgrade my NixOS installation to current nixos-unstable (2023-10-24) but the build fails with the following error:

```nix
$ nix --extra-experimental-features nix-command flakes build .#nixosConfigurations."optibox".config.system.build.toplevel --verbose --out-link /tmp/nixos-rebuild.jKcMUv/result
error:
       … while calling the 'head' builtin

         at /nix/store/z1nvpjx9vd4151vx2krxzmx2p1a36pf9-source/lib/attrsets.nix:850:11:

          849|         || pred here (elemAt values 1) (head values) then
          850|           head values
             |           ^
          851|         else

       … while evaluating the attribute 'value'

         at /nix/store/z1nvpjx9vd4151vx2krxzmx2p1a36pf9-source/lib/modules.nix:807:9:

          806|     in warnDeprecation opt //
          807|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          808|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: getting status of '/nix/store/fijy0dlyk5b06ay7nxg88lr7bp13hpjb-iproute2-6.5.0/etc': No such file or directory
```

Seems like iproute2's `etc` is missing? 

I'm not sure what other data might be helpful. 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11.20231019.7c9cc5a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source`
```

**Comment by spacekitteh** [comment]
Same for me.

**Comment by spacekitteh** [comment]
The fix is in https://github.com/NixOS/nixpkgs/pull/262767

**Comment by naturallaw777** [comment]
@spacekitteh how did you fix this?

**Comment by ParetoOptimalDev** [comment]
> @spacekitteh how did you fix this?

It looks like it'll probably be in unstable soon.

It doesn't seem to have worked or I have some other bug, but you may have luck with what I tried doing:

### use a derivation directly from a nixpkgs pull request

#### add input

```nix
nixpkgs-iproute2-ifd.url = "github:nixos/nixpkgs/pull/242453/head";
```

#### taking advantage of lexical scoping to get at flake inputs from overlay (not sure how to do this without)

```nix
nixpkgs.overlays = [
  (final: prev: {
    iproute2 = flake.inputs.nixpkgs-iproute2-fix.legacyPackages.${final.stdenv.system}.iproute2;
  })
];

**Comment by naturallaw777** [comment]
I had to disable the Mullvad module by commenting out ```# services.mullvad-vpn.enable = true;```
Next, I updated to the latest nixpkgs NixOS 23.11.20231024.8efd5d1 (Tapir) x86_64 and it worked with no errors.
Then, I un-commented Mullvad module to activate it again. Then, I rebuilt switch and the error returned.

Thus, it seems the Mullvad module needs to be updated.

**Comment by purcell** [comment]
@naturallaw777 I think that the mullvad module [enables the iproute2 module](https://github.com/NixOS/nixpkgs/blob/df67798db1bd148f44efcb47ee3da094ba4f2fcc/nixos/modules/services/networking/mullvad-vpn.nix#L42), so is it perhaps that without mullvad, you're not requiring `iproute2` and so there's no problem?

**Comment by naturallaw777** [comment]
@purcell thanks for thoughts on this... I don't have ```iproute2``` module explicitly in my ```configuration.nix```. Is there something I am missing regarding the logic?

Should I explicitly put the ```iproute2``` in my ```configuration`` to force the update?

**Comment by naturallaw777** [comment]
Yes, it is not Mullvad per say. It is the broken iproute2 module  (```networking.iproute2.enable = true;```).  I will wait until it is merged into the main unstable branch for the fix.

**Comment by purcell** [comment]
@naturallaw777 this link might help you: https://nixpk.gs/pr-tracker.html?pr=262767

**Comment by howdoicomputer** [comment]
I feel like every time I run `nixos-rebuild` something upstream breaks.

**Comment by nadir-yttrium** [comment]
> I feel like every time I run `nixos-rebuild` something upstream breaks.

The issue tracker is the wrong place for useless comments like that.

**Comment by IogaMaster** [comment]
> > @spacekitteh how did you fix this?
> 
> It looks like it'll probably be in unstable soon if you can wait.
> 
> It doesn't seem to have worked or I have some other bug, but you may have luck with what I tried doing:
> 
> ### use a derivation directly from a nixpkgs pull request
> #### add input
> ```nix
> nixpkgs-iproute2-ifd.url = "github:nixos/nixpkgs/pull/242453/head";
> ```
> 
> #### taking advantage of lexical scoping to get at flake inputs from overlay (not sure how to do this without)
> ```nix
> nixpkgs.overlays = [
>   (final: prev: {
>     iproute2 = flake.inputs.nixpkgs-iproute2-fix.legacyPackages.${final.stdenv.system}.iproute2;
>   })
> ];
> ```

For anyone having an issue with this temp fix:


 #### add input
 ```nix
 nixpkgs-iproute2-fix.url = "github:nixos/nixpkgs/pull/242453/head";
 ```
 #### taking advantage of lexical scoping to get at flake inputs from overlay (not sure how to do this without)
 ```nix
 nixpkgs.overlays = [
   (final: prev: {
     iproute2 = flake.inputs.nixpkgs-iproute2-fix.legacyPackages.${final.stdenv.system}.iproute2;
   })
 ];

He made a typo, the input is called `nixpkgs-iproute2-ifd` the overlay calls for `nixpkgs-iproute2-fix`
Just change the input to `nixpkgs-iproute2-fix`

Thanks to @ardishco-the-great, he brought this issue to my attention

**Comment by IogaMaster** [comment]
@ParetoOptimalDev thanks for the fix, keep up the good work! :+1:

**Comment by purcell** [comment]
@IogaMaster You should probably prefer to overlay the merged fix PR #262767, not the unmerged #242453.

**Comment by purcell** [comment]
The fix PR has hit `nixos-unstable`, and works great on my machine, for those who were waiting.

**Comment by IogaMaster** [comment]
> @IogaMaster You should presumably prefer to overlay the merged fix PR #262767, not the unmerged #242453, no?

Yeah, I was just fixing the typo
